chore(logs) update alerting and playbooks (#731)
cloudoperators/greenhouse-extensions: logs/playbooks/OTelLogsMissing.md

---
title: Logs are Missing
weight: 20
---

## Root Cause Analysis

Determine if the lack of sent logs is expected:
- Check if the Plugin is up and running via the Greenhouse Dashboard.
- Check if there are other alerts that would indicate a lack of sent logs (CrashLoopBackoff, ErrImagePull).
- Check operational logs for the Pods (requests failing, incorrect credentials for logshipping).

> - Check that the backend (e.g. OpenSearch) is ready to receive logs.

Reviewer: How does one know/check that OS is ready?

## Solution

### 1. In Case of Expected Downtime
Mute the alert temporarily via Greenhouse until the plugin is healthy again and notify the team responsible service owner.

### 2. In Case of an Unexpected Downtime

> #### 2.1 If it is a connection error and other Pods are working, investigate the errors by checking the Pods in the designated namespace:

Reviewer: ```suggestion
#### 2.1 Are other Pods (not 'logs-collector' ones) working?

If yes, it could be a connection error. Investigate it by checking the errors of the 'logs-collector' Pods in the designated namespace:
```

```
# Retrieves all pods, find the failing pod.

> kubectl  k get pods -n <namespace> -o wide | grep -e 'collector'

Reviewer: ```suggestion
kubectl get pods -n <namespace> -o wide | grep -e 'collector'
```

```

> #### 2.2 You can also check the logs of the associated underlying container for more errors by running:

Reviewer: ```suggestion
#### 2.2 If you find a problematic pod, errors in the log outputs, or other issues. You should also check the logs of a particular _container_ of the problematic Pod for errors, by running:
```

```

> kubectl logs <failing-logs-collector> -c otc-container

Reviewer: ```suggestion
kubectl logs <failing-logs-collector> -c otc-container -n <namespace>
```

```

> #### 2.3 Delete the affected OpenTelemetry Pods in logs namespace of the controlplane Kubernetes.

Reviewer: ```suggestion
#### 2.3 Delete the affected OpenTelemetry Pods in the namespace of the controlplane.
```

```
# Delete a specific pod

> kubectl  delete pod <failing-logs-collector> -n logs

Reviewer: ```suggestion
kubectl  delete pod <failing-logs-collector> -n <namespace>
```
